public/app/features/explore/QueryRowActions.test.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { QueryRowActions, Props } from './QueryRowActions';
+import { shallow } from 'enzyme';
+
+const setup = (propOverrides?: object) => {
+  const props: Props = {
+    hideQuery: true,
+    canToggleEditorModes: true,
+    onClickToggleEditorMode: () => {},
+    onClickToggleHiddenQuery: () => {},
+    onClickAddButton: () => {},
+    onClickRemoveButton: () => {},
+  };
+
+  Object.assign(props, propOverrides);
+
+  const wrapper = shallow(<QueryRowActions {...props} />);
+  return wrapper;
+};
+
+describe('QueryRowActions', () => {
+  it('should render component', () => {
+    const wrapper = setup();
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('should render component without editor mode', () => {
+    const wrapper = setup({ canToggleEditorModes: false });
+    expect(wrapper.find('i.fa-pencil')).toHaveLength(0);
```

Comment:
One way to also do this to check for elements is to use `aria-labels`. Those should be more stable (ie `query edit` label should be more stable than particular icon class). Not sure if it is applicable later in the show/hide button tests if we want to test if icon actually changes but in general when finding some particular element in the DOM, `aria-label` is preferred to classes or IDs, and also has the benefit of better accessibility.